src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -190,6 +190,22 @@ typedef struct WorkerShardStatistics
 	HTAB *statistics;
 } WorkerShardStatistics;
 
+/* ShardMoveDependencyHashKey contains colocationId to be used as a key in a hash */
+typedef struct ShardMoveDependencyHashKey
+{
+	int64 colocationId;
+} ShardMoveDependencyHashKey;
+
+/* ShardMoveDependencyHashEntry contains the taskId which any new shard move task within the corresponding colocation group must take a dependency on */
+typedef struct ShardMoveDependencyInfo
+{
+	ShardMoveDependencyHashKey key;
+	int64 taskId;
+} ShardMoveDependencyInfo;
+
+HTAB *colocationDependencyHashMap = NULL;
+HTAB *nodeDependencyHashMap = NULL;
```

Comment:
We try to avoid global variables when reasonably possible, it makes it more easy to reason about what a function its inputs are. To avoid having to pass multiple related arguments around, we often wrap such arguments in a struct definition like this: Something like this:

```c
typedef struct ShardMoveDependencies
{
	HTAB *colocationDependencies;
	HTAB *nodeDependencies;
} ShardMoveDependencies;
```

P.S. When we do use global variables we make the first letter a capital, to distinguish it easily from local variables.